@react-pdf/reconciler tries to access unlisted dependencies

**Describe the bug**

@react-pdf/reconciler tries to access unlisted dependencies, making the installations non-deterministic and depending on pure luck to work.

**To Reproduce**

Install @react-pdf using Yarn PnP, which points out such errors immediately instead of failing silently.

**Expected behavior**

Package to work

**Actual behavior**

```
✘ [ERROR] Could not resolve "react"

    ../~/.yarn/berry/cache/@react-pdf-reconciler-npm-1.1.2-7ab8edc928-10c0.zip/node_modules/@react-pdf/reconciler/lib/index.js:1:18:
      1 │ import React from 'react';
        ╵                   ~~~~~~~

  The Yarn Plug'n'Play manifest forbids importing "react" here because it's not listed as a dependency of this package:

    ../.pnp.cjs:2880:31:
      2880 │         "packageDependencies": [\
           ╵                                ~~
```

```
✘ [ERROR] Could not resolve "object-assign"

    ../~/.yarn/berry/cache/@react-pdf-reconciler-npm-1.1.2-7ab8edc928-10c0.zip/node_modules/@react-pdf/reconciler/lib/reconciler-26.js:1:13:
      1 │ import e from"object-assign";import n from"react";import*as t from"...
        ╵              ~~~~~~~~~~~~~~~

  The Yarn Plug'n'Play manifest forbids importing "object-assign" here because it's not listed as a dependency of this package:

    ../.pnp.cjs:2880:31:
      2880 │         "packageDependencies": [\
           ╵                                ~~
```

```
✘ [ERROR] Could not resolve "scheduler"

    ../~/.yarn/berry/cache/@react-pdf-reconciler-npm-1.1.2-7ab8edc928-10c0.zip/node_modules/@react-pdf/reconciler/lib/reconciler-26.js:1:66:
      1 │ ...rom"react";import*as t from"scheduler";function r(e){return e&&e...
        ╵                               ~~~~~~~~~~~

  The Yarn Plug'n'Play manifest forbids importing "scheduler" here because it's not listed as a dependency of this package:

    ../.pnp.cjs:2880:31:
      2880 │         "packageDependencies": [\
           ╵                                ~~
```

Hello have the same problem:
* react-pdf/renderer@4.1.2
* next@15.0.3
* react@18.3.1


`
./node_modules/@react-pdf/reconciler/lib/reconciler-26.js:1:51
Module not found: Can't resolve 'scheduler'
`

--- a/.changeset/wicked-apes-kick.md
+++ b/.changeset/wicked-apes-kick.md
@@ -1,0 +1,5 @@
+---
+"@react-pdf/reconciler": patch
+---
+
+fix(reconciler): missing dependencies

--- a/packages/reconciler/package.json
+++ b/packages/reconciler/package.json
@@ -22,11 +22,17 @@
   "files": [
     "lib"
   ],
+  "peerDependencies": {
+    "react": "^16.8.0 || ^17.0.0 || ^18.0.0 || ^19.0.0"
+  },
+  "dependencies": {
+    "object-assign": "^4.1.1",
+    "scheduler": "0.25.0-rc-603e6108-20241029"
+  },
   "devDependencies": {
     "ast-types": "^0.14.2",
     "react-reconciler-26": "npm:react-reconciler@0.26.0",
     "react-reconciler-31": "npm:react-reconciler@0.31.0-rc-603e6108-20241029",
-    "scheduler": "0.25.0-rc-603e6108-20241029",
     "recast": "^0.23.9"
   }
 }

